java-manta-client/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1478,61 +1479,127 @@ public void putSnapLink(final String rawLinkPath, final String rawObjectPath,
     }
 
     /**
-     * Moves a file from one path to another path. This operation is not
-     * transactional and will fail or produce inconsistent result if the source
-     * or the destination is modified while the operation is in progress.
+     * <p>Moves an object from one path to another path. When moving
+     * directories or files between different directories, this operation is
+     * not transactional and may fail or produce inconsistent result if
+     * the source or the destination is modified while the operation is
+     * in progress.</p>
+     *
+     * <p><em>Note: If you need to do a rename, then use
+     * {@link #putSnapLink(String, String, MantaHttpHeaders)}</em></p>
      *
      * @param source Original path to move from
      * @param destination Destination path to move to
      * @throws IOException thrown when something goes wrong
      */
     public void move(final String source, final String destination)
             throws IOException {
+        move(source, destination, false);
+    }
+
+    /**
+     * <p>Moves an object from one path to another path. When moving
+     * directories or files between different directories, this operation is
+     * not transactional and may fail or produce inconsistent result if
+     * the source or the destination is modified while the operation is
+     * in progress.</p>
+     *
+     * <p><em>Note: If you need to do a rename, then use
+     * {@link #putSnapLink(String, String, MantaHttpHeaders)}</em></p>
+     *
+     * @param source Original path to move from
+     * @param destination Destination path to move to
+     * @param recursivelyCreateDestinationDirectories when true create the full destination directory path
+     * @throws IOException thrown when something goes wrong
+     */
+    public void move(final String source, final String destination,
+                     final boolean recursivelyCreateDestinationDirectories)
+            throws IOException {
         LOG.debug("Moving [{}] to [{}]", source, destination);
 
-        MantaObjectResponse entry = head(source);
+        final String formattedSource = formatPath(source);
+        final String formattedDestination = formatPath(destination);
```

Comment:
urlencoding is sadness and I'm afraid my previous comment was misleading.  Since urlencoding is not idempotent it must be done only once.   For example,  if we encode in `move()` the path will be double encoded on calls to `putDirectory`, resulting in erroneous object names.  When I did the last round of encoding changes I tried to differentiate between the `path` (encoded) and `rawPath` (not-encoded) as parameter names, but I didn't go to change methods that were not broken and I'm not sure how much that clarifies the situation.   This is all terribly error prone and maybe it might be better to have a `MantaPath` object if we redo the api yet again.

Anyway, what I was worried about was some corner case where `FilenameUtils.getFullPath` would have surprising results relative to the later encoded path.  I don't think there is one but am not familiar with that code.  Sorry for leading you astray!

I did notice while looking at the docs again that `FilenameUtils.getFullPath` can return `null`.  I'm not sure what they mean by an invalid path, but based on our recent dealings with exceptions-lacking-context we might want to check for that to error in a more useful way.